web/src/components/AssertionCard/AssertionCard.tsx

```diff
@@ -1,32 +1,26 @@
 import {useCallback, useMemo} from 'react';
 import {useStore} from 'react-flow-renderer';
-import {TAssertionResultEntry, TSpanSelector} from '../../types/Assertion.types';
+import {TAssertionResultEntry} from '../../types/Assertion.types';
 import AssertionCheckRow from '../AssertionCheckRow';
 import * as S from './AssertionCard.styled';
 
 interface TAssertionCardProps {
   assertionResult: TAssertionResultEntry;
-  selectorList: TSpanSelector[];
   onSelectSpan(spanId: string): void;
-  onDelete(assertionId: string): void;
+  onDelete(selector: string): void;
   onEdit(assertionResult: TAssertionResultEntry): void;
 }
 
 const AssertionCard: React.FC<TAssertionCardProps> = ({
-  assertionResult: {resultList, id: assertionId},
-  selectorList,
+  assertionResult: {selector, resultList, selectorList, pseudoSelector, spanCount},
   assertionResult,
   onSelectSpan,
   onDelete,
   onEdit,
 }) => {
   const store = useStore();
 
-  const spanCountText = useMemo(() => {
-    const spanCount = resultList.length;
-
-    return `${spanCount} ${spanCount > 1 ? 'spans' : 'span'}`;
-  }, [resultList.length]);
+  const spanCountText = useMemo(() => `${spanCount} ${spanCount > 1 ? 'spans' : 'span'}`, [spanCount]);
```

Comment:
I don't think `useMemo` has an advantage here. This seems to be a simple operation. Remember that `useMemo`, `useCallback` also has a cost in performance.